doc/rst/usingchapel/chplenv.rst

```diff
@@ -417,10 +417,15 @@ CHPL_LOCALE_MODEL
         flat     top-level locales are not further subdivided
         numa     top-level locales are further subdivided into
```

Comment:
@ShreyasKhandekar : Not your PR so arguably not your problem, but the numa locale model has been deprecated and is no longer tested, so I'm wondering whether you'd be up for removing this entry from the table?  (which caught my eye while checking the diff on this change).  This would also have the side advantage of re-alphabetizing this table.  :D

Grepping the docs, it looks like this and `rst/usingchapel/multilocale.rst` are the two places it could be removed.

Reply:
Sure!

Reply:
https://github.com/chapel-lang/chapel/pull/25333 will do this. `rst/usingchapel/multilocale.rst` didn't talk about the numa locale mode, rather how to use the `-nl NxLt` for launching chapel programs based on numa domains, so I left it as is.

Reply:
Thanks, Shreyas!  (and for catching my mistake)